content/blog/2022-06-22-azure-remotes-in-dvc.md

```diff
@@ -0,0 +1,231 @@
+tracking experiments. Now we need to set up the remote to connect our project
+data stored in Azure to the DVC repo.
+
+### Create a default remote
+
+Now we can add a default to the project with the following command:
+
+```dvc
+$ dvc remote add -d bikes azure://bikedata
+```
+
+This creates a default remote called `bikes` that connects to the `bikedata`
+container we made earlier which is where the training data for the model will be
+stored.
+
+### Add Azure credentials
+
+In order for DVC to be able to push and pull data from the remote, you need to
+have valid Azure credentials.
+
+By default, DVC authenticates using your
+[Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+configuration.
+
+Run the following command to authenticate with Azure.
+
+```dvc
+$ az login
+A web browser has been opened at https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize. Please continue the login in the web browser. If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.
+[
+  {
+    "cloudName": "AzureCloud",
+    "homeTenantId": "some-id",
+    "id": "some-id",
+    "isDefault": true,
+    "managedByTenants": [],
+    "name": "Azure subscription 1",
+    "state": "Enabled",
+    "tenantId": "some-id",
+    "user": {
+      "name": "test@test.com",
+      "type": "user"
+    }
+  }
+]
+```
+
+This should open a window that looks like this where you can enter your login
+credentials.
+
+![Azure CLI authentication page](/uploads/images/2022-06-22/azure_auth_page.png)
+
+You can check out more details on this command
+[here in the Azure docs](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli).
+If you want to use a different authentication method with DVC, check out
+[our docs here](https://dvc.org/doc/command-reference/remote/modify#microsoft-azure-blob-storage).
+
+You will also need to manually define the storage account name with the
+following command:
+
+```dvc
+$ dvc remote modify bikes account_name 'bicycleproject'
+```
+
+### Push and pull data with DVC
+
+Now you can push data from your local machine to the Azure remote! First, add
+the data you want DVC to track with the following command:
+
+```dvc
+$ dvc add data
+```
+
+This will allow DVC to track the entire `data` directory so it will note when
+any changes are made. Then you can push that data to your Azure remote with this
+command:
+
+```dvc
+$ dvc push
+```
+
+Here's what the data might look like in your Azure container.
+
+![data in Azure container](/uploads/images/2022-06-22/data_in_azure.png)
+
+Then if you move to a different machine or someone else needs to use that data,
+it can be accessed by cloning or forking the project repo and running:
+
+```dvc
+$ dvc pull
+```
+
+This will get any data from your remote and download it to your local machine.
+
+<admon type="info">
+
+Authentication has to be setup locally on any machine you need to pull or push
+data from.
```

Comment:
Does this mean just `az login`? It might be better to specify that so it's clear you don't have to redo the `dvc remote` commands or anything.